### PR TITLE
[SPARK-49232][INFRA][TESTS] Remove `3.4.2` from `SKIP_SPARK_RELEASE_VERSIONS`

### PR DESCRIPTION
**Pull request by panbingkun**
### What changes were proposed in this pull request?
The pr aims to remove `3.4.2` from `SKIP_SPARK_RELEASE_VERSIONS(build_maven.yml)`.

### Why are the changes needed?
- Why was `SKIP_SPARK_RELEASE_VERSIONS` introduced? See here: https://issues.apache.org/jira/browse/SPARK-46400

- [SPARK-46400](https://github.com/apache/spark/pull/45017) has already merged into Spark `3.4.3`, and the corresponding issue no longer exists for Spark `3.4.3+`

- The following location only includes `the version` we have `fixed` for this issue
  https://dist.apache.org/repos/dist/release/spark/
  image

### Does this PR introduce _any_ user-facing change?
No, only for GA's tests.

### How was this patch tested?
Pass GA.

### Was this patch authored or co-authored using generative AI tooling?
No.

### CODE REVIEW

**Comment by panbingkun** [comment]
cc @dongjoon-hyun @LuciferYang

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala [hunk 1]
```diff
@@ -268,8 +262,7 @@ object PROCESS_TABLES extends QueryTest with SQLTestUtils {
         .filter(_.contains("""<a href="spark-"""))
         .filterNot(_.contains("preview"))
         .map("""<a href="spark-(\d.\d.\d)/">""".r.findFirstMatchIn(_).get.group(1))
-        .filter(_ < org.apache.spark.SPARK_VERSION)
```
Comment (LuciferYang) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala:271]: Maybe we can keep this function. I am not sure if there will be similar needs in the future that allow us to reuse it again.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala [hunk 2]
```diff
@@ -268,8 +262,7 @@ object PROCESS_TABLES extends QueryTest with SQLTestUtils {
...
-        .filterNot(skipReleaseVersions.contains).toImmutableArraySeq
```
Comment (LuciferYang) [inline sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala:272]: Maybe we can keep this function. I am not sure if there will be similar needs in the future that allow us to reuse it again.

Comment (panbingkun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala:272]: Okay

Comment (panbingkun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala]: Updated.

Comment (LuciferYang) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala]: We should update pr title and pr description to match the change

**Review by dongjoon-hyun — COMMENTED** [review]
Thank you for revision according to @LuciferYang 's comment.

Could you revise the PR title and description as suggested, @panbingkun ?

**Review by LuciferYang — APPROVED** [review]

**Comment by panbingkun** [comment]
> Thank you for revision according to @LuciferYang 's comment.
> 
> Could you revise the PR title and description as suggested, @panbingkun ?

Oh, I'm sorry, I miss to synchronize this update (Jira's title has also been updated). 
It's now completed.

File: sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala (hunk 2 above)
Comment (panbingkun) [reply sql/hive/src/test/scala/org/apache/spark/sql/hive/HiveExternalCatalogVersionsSuite.scala]: Yea, updated.

**Review by dongjoon-hyun — APPROVED** [review]

**Comment by dongjoon-hyun** [comment]
Merged to master~

**Pull request closed without merging**